packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { BareProps, I18nProps } from './types';
+
+import BN from 'bn.js';
+import React from 'react';
+import isString from '@polkadot/util/is/string';
+
+import classes from './util/classes';
+import Input, { KEYS, KEYS_PRE, isCopy, isCut, isPaste, isSelectAll } from './Input';
+import translate from './translate';
+
+type Props = BareProps & I18nProps & {
+  defaultValue?: string,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
```

Comment:
I believe we need to pass in bitLength here instead of maxLength. Additionally, put that at 32-bits default (i.e. which is just "normal" numbers, for Balance and friends we will pass through the correct length as required.